poetry/core/masonry/builders/wheel.py

```diff
@@ -91,44 +96,59 @@ def build(self):
 
     def _build(self, wheel):
         if self._package.build_script:
-            with SdistBuilder(poetry=self._poetry).setup_py() as setup:
-                # We need to place ourselves in the temporary
-                # directory in order to build the package
+            if not self._poetry.package.build_config.get("generate-setup-file", True):
+                # Since we have a build script but no setup.py generation is required,
+                # we assume that the build script will build and copy the files
+                # directly.
```

Comment:
Would this be a problematic assumption? Aren't we encouraging different behaviour based on an unrelated configuration? ie. this could break existing build scripts if the maintainer chooses to switch of setup file generation (or when we switch to it being disabled by default).

Reply:
The `build` script feature of Poetry was never part of the public API and an undocumented feature up until now since I knew from the start that we would no longer rely on `setup.py` at some point and that the current implementation was a hack more than anything. So breaking this does not feel too problematic to me. Also I don't think a lot of people relies on it at the moment.

Delegating the bulk of the work to the `build` script was always the plan so that people can build their extensions any way they see fit, without being tied to a specific build tool like `setuptools`.